spec/museum_spec.rb

```diff
@@ -0,0 +1,196 @@
+            imax = Exhibit.new({name: "IMAX",cost: 15})
+
+            dmns.add_exhibit(gems_and_minerals)
+            dmns.add_exhibit(dead_sea_scrolls)
+            dmns.add_exhibit(imax)
+
+            patron_1 = Patron.new("Bob", 0)
+            patron_1.add_interest("Gems and Minerals")
+            patron_1.add_interest("Dead Sea Scrolls")
+
+            patron_2 = Patron.new("Sally", 20)
+            patron_2.add_interest("Dead Sea Scrolls")
+
+            patron_3 = Patron.new("Johnny", 5)
+            patron_3.add_interest("Dead Sea Scrolls")
+
+            dmns.admit(patron_1)
+            dmns.admit(patron_2)
+            dmns.admit(patron_3)
+
+            expect(dmns.ticket_lottery_contestants(dead_sea_scrolls)).to eq([patron_1, patron_3])
+            expect(dmns.ticket_lottery_contestants(gems_and_minerals)).to eq([])
+            expect(dmns.ticket_lottery_contestants(imax)).to eq([])
+        end
+    end
+
+    describe '#draw_lottery_winner' do
+        it 'returns a random winner from the eligible lottery contestants' do
+            dmns = Museum.new("Denver Museum of Nature and Science")
+            gems_and_minerals = Exhibit.new({name: "Gems and Minerals", cost: 0})
+            dead_sea_scrolls = Exhibit.new({name: "Dead Sea Scrolls", cost: 10})
+            imax = Exhibit.new({name: "IMAX",cost: 15})
+
+            dmns.add_exhibit(gems_and_minerals)
+            dmns.add_exhibit(dead_sea_scrolls)
+            dmns.add_exhibit(imax)
+
+            patron_1 = Patron.new("Bob", 0)
+            patron_1.add_interest("Gems and Minerals")
+            patron_1.add_interest("Dead Sea Scrolls")
+
+            patron_2 = Patron.new("Sally", 20)
+            patron_2.add_interest("Dead Sea Scrolls")
+
+            patron_3 = Patron.new("Johnny", 5)
+            patron_3.add_interest("Dead Sea Scrolls")
+
+            dmns.admit(patron_1)
+            dmns.admit(patron_2)
+            dmns.admit(patron_3)
+
+            winner = dmns.draw_lottery_winner(dead_sea_scrolls)
+
+            expect([patron_1.name, patron_3.name]).to include(winner) #could have maybe stubbed here
+        end
+
+        it 'returns nil if there are no patrons eligible' do
+            dmns = Museum.new("Denver Museum of Nature and Science")
+            gems_and_minerals = Exhibit.new({name: "Gems and Minerals", cost: 0})
+
+            dmns.add_exhibit(gems_and_minerals)
+
+            expect(dmns.draw_lottery_winner(gems_and_minerals)).to be_nil #No patrons admitted, making sure functioning properly
+        end
+    end
+
+    describe '#announce_lottery_winner' do
+        it 'announces lotto winner if there is one' do
+            dmns = Museum.new("Denver Museum of Nature and Science")
+            imax = Exhibit.new({name: "IMAX",cost: 15})
+
+            dmns.add_exhibit(imax)
+
+            allow(dmns).to receive(:draw_lottery_winner).and_return("Bob") #Do I need to specify imax somehow?
```

Comment:
You do not need to specify anything about IMAX - its not important to the functionality of this method :) GREAT JOB!!!!